search/indexing_api.py

```diff
@@ -109,12 +110,16 @@ def index_program_enrolled_users(program_enrollments, chunk_size=100):
     Returns:
         int: Number of indexed items
     """
-    return _index_chunks(
+    num_indexed = _index_chunks(
         (serialize_program_enrolled_user(program_enrollment) for program_enrollment in program_enrollments),
         USER_DOC_TYPE,
         chunk_size=chunk_size,
     )
 
+    for program_enrollment in program_enrollments:
+        send_automatic_emails(program_enrollment)
```

Comment:
If we still care about indexing performance (and I believe we do https://github.com/mitodl/micromasters/issues/2746), I think we should figure out a different approach for this. I think I'm mostly hung up on the fact that this indexing method (`index_program_enrolled_users`) is invoked from multiple sources, including `recreate_index`. It seems like it's just being added here for convenience. From what I can tell, this works for our purposes because `dashboard.signals.handle_create_programenrollment` calls `index_program_enrolled_users`, so the effect is that a new program enrollment is created for a user, and the email is sent (if appropriate). On the other hand I don't think it makes any sense for us to attempt to send automatic emails as part of a complete recreation of the index, but that's what will happen. 

I think it will make more sense to trigger the automatic email send only when a new program enrollment is created. `dashboard.signals.handle_create_programenrollment` calls `search.tasks.index_program_enrolled_users` right now, but maybe it makes more sense to create a new task called `index_new_program_enrolled_user`, which would call `search.indexing_api.index_program_enrolled_users` _and then_ attempt to send the automatic email via some other function. Does that make sense?